File: Day310/sumOfAbsolutionDifferencesInASortedArray.java
class Solution {
    public int[] getSumAbsoluteDifferences(int[] nums) {
        int arr[] = new int[nums.length];
        int total = 0;

        for (int i=0; i<nums.length; i++) {
            total += nums[i];    
        }
        int l = 0, r = total;
        for (int i=0; i<nums.length; i++) {
            r -= nums[i];
            arr[i] = (r - l + (i*nums[i]) - (nums.length- i - 1) * nums[i]);
            l += nums[i];
        }
        return arr;
    }
}

File: Day310/sumOfAbsolutionDifferencesInASortedArray.go
func getSumAbsoluteDifferences(nums []int) []int {
    answer := make([]int ,0)
    total := 0
    for _,v := range nums {
        total += v
    }
    pre, rem := 0, total
    for i,v := range nums {
        rem -= v
        answer = append(answer, rem - pre + (i *v) - ((len(nums)-i-1) * v))
        pre += v
    }
    return answer
}